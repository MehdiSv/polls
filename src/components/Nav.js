import { Link } from "react-router-dom";
import { connect } from "react-redux";

const Nav = ({ users, authedUser }) => {
  return (
    <nav className="nav">
      <ul>
        <li>
          <img
            src={users[authedUser].avatarURL}
            alt={`Avatar of ${users[authedUser].name}`}
            className="avatar"
          />
          {authedUser}
        </li>
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link data-testid="leaderbord-link" to="/leaderboard">
            Leaderboard
          </Link>
        </li>
        <li>
          <Link data-testid="add-poll-link" to="/add">
            New
          </Link>
        </li>
        <li>
          <Link to="/logout">Logout</Link>
        </li>
      </ul>
    </nav>
  );
};

const mapStatetoProps = ({ users, authedUser }) => {
  return {
    users,
    authedUser,
  };
};

export default connect(mapStatetoProps)(Nav);
