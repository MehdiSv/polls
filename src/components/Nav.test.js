import { render } from "@testing-library/react";
import Nav from "./Nav";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter } from "react-router-dom";

const users = {
  sarahedo: {
    id: "sarahedo",
    password: "password123",
    name: "Sarah Edo",
    avatarURL:
      "https://gravatar.com/avatar/14299fa7ecd7f50203ee5c4c1d60f446?s=400&d=robohash&r=x",
    answers: {
      "8xf0y6ziyjabvozdd253nd": "optionOne",
      "6ni6ok3ym7mf1p33lnez": "optionOne",
      am8ehyc8byjqgar0jgpub9: "optionTwo",
      loxhs1bqm25b708cmbf3g: "optionTwo",
    },
    questions: ["8xf0y6ziyjabvozdd253nd", "am8ehyc8byjqgar0jgpub9"],
  },
};

const authedUser = "sarahedo";

const mockStore = configureStore([]);

const store = mockStore({ users, authedUser });

describe("Nav", () => {
  it("will match snapshot", () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it("contains the leaderboard and new links", () => {
    var component = render(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>
    );

    expect(component.getByTestId("leaderbord-link")).toBeInTheDocument();
    expect(component.getByTestId("add-poll-link")).toBeInTheDocument();
  });
});
