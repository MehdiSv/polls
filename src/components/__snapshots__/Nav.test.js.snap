// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <ul>
          <li>
            <img
              alt="Avatar of Sarah Edo"
              class="avatar"
              src="https://gravatar.com/avatar/14299fa7ecd7f50203ee5c4c1d60f446?s=400&d=robohash&r=x"
            />
            sarahedo
          </li>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              data-testid="leaderbord-link"
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </li>
          <li>
            <a
              data-testid="add-poll-link"
              href="/add"
            >
              New
            </a>
          </li>
          <li>
            <a
              href="/logout"
            >
              Logout
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <ul>
        <li>
          <img
            alt="Avatar of Sarah Edo"
            class="avatar"
            src="https://gravatar.com/avatar/14299fa7ecd7f50203ee5c4c1d60f446?s=400&d=robohash&r=x"
          />
          sarahedo
        </li>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            data-testid="leaderbord-link"
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
        <li>
          <a
            data-testid="add-poll-link"
            href="/add"
          >
            New
          </a>
        </li>
        <li>
          <a
            href="/logout"
          >
            Logout
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
